[WC-406] Accordion widget (web) - Add functional & Atlas styles

IIsaku opened the pull request:
## Checklist
- Contains unit tests ✅ 
- Contains breaking changes ❌
- Contains Atlas changes ✅ 
- Compatible with: MX 9️⃣

#### Web specific
- Contains e2e tests ❌
- Is accessible ✅ 
- Compatible with Studio ✅ 
- Cross-browser compatible ✅ 

#### Feature specific
- Comply with designs ✅ 
- Comply with PM's requirements ✅ 

## This PR contains
- [ ] Bug fix
- [x] Feature
- [ ] Refactor
- [ ] Documentation
- [ ] Other (describe)

## What is the purpose of this PR?
Introduce functional & Atlas styles to the accordion widget.

## Relevant changes
- Add specific classes to each accordion DOM element.
- Add "Dynamic class" widget property for each accordion group to apply Atlas brand/semantic styles.
- Add Studio Pro widget icon for the accordion widget.
- Add functional styles to the accordion widget.
- Add default & helper styles to Atlas for the accordion widget.
- Add specific Atlas variables for the accordion widget.

## What should be covered while testing?
- Functional widget styling (without Atlas).
- Atlas styling including brand styles with the help of the "Dynamic class" property.
- Accordion variables inside Atlas.

IIsaku committed:
feat(accordion-web): add functional styles

IIsaku committed:
feat(accordion-web): extend functional styles

- Add specific class to each HTML element to simplify styling.
- Restructure and add styles.

IIsaku committed:
feat(accordion-web): add Studio Pro icon

IIsaku committed:
feat(accordion-web): add functional and default Atlas styles

- Improve functional styles.
- Add Atlas default widget styles and setup helper classes.
- Nest all Atlas widget styles inside the root class, so that
the styles have precedence over the inline styles.
- Add exclusion variables for Accordion styles.
- Add Accordion specific variables that point to button styles by default.

IIsaku committed:
feat(accordion-web): add dynamic class property per group

IIsaku committed:
feat(atlas-ui): add brand styles

diego-antonelli commented on packages/pluggableWidgets/accordion-web/src/ui/accordion-main.scss:
```diff
@@ -0,0 +1,48 @@
+$border-color: #ced0d3;
+
+.widget-accordion {
+    background-color: #ffffff;
```
Can we use a variable for the background too?

IIsaku replied:
Yes, I also added variables for the other colors that are used in the functional styles.

diego-antonelli commented on packages/pluggableWidgets/accordion-web/src/ui/accordion-main.scss:
```diff
@@ -0,0 +1,48 @@
+    .widget-accordion-group-header {
+        padding: 16px;
+
+        > :is(h1, h2, h3, h4, h5, h6, p) {
```
Nice one

IIsaku replied:
Thank you 😊

diego-antonelli commented on packages/pluggableWidgets/accordion-web/src/ui/accordion-main.scss:
```diff
@@ -0,0 +1,48 @@
+
+        &.widget-accordion-group-header-clickable {
+            &:hover,
+            &:hover *,
```
I think this might cause some weird behavior in some widgets if combined, test with a badge inside the header to see what happens

IIsaku replied:
Fixed in both functional and Atlas styles.

IIsaku committed:
feat(atlas-ui): add accordion border design property

IIsaku committed:
feat(atlas-ui): add accordion striped design property

IIsaku committed:
feat(atlas-ui): add accordion compact design property

IIsaku committed:
refactor(accordion-web): add color variables to functional styles

IIsaku committed:
fix: functional & Atlas header hover styles

IIsaku committed:
fix: functional & Atlas header hover styles

IIsaku committed:
chore(accordion-web): update unit test snapshots due to added classnames

IIsaku committed:
refactor(atlas-ui): remove duplicate styles related to functional styles

IIsaku committed:
feat: improve styles

- Rename borders design property "vertical" value to "horizontal".
- Rename accordion variables in Atlas.
- Improve functional & Atlas accordion styles.

IIsaku committed:
fix: accordion border styles

IIsaku committed:
feat(atlas-ui): improve semantic styles with striped design property

IIsaku committed:
feat(atlas-ui): improve Atlas styles

- Improve accordion design property descriptions.
- Refactor design property helper styles.

IIsaku committed:
chore(accordion-web): update test project

IIsaku committed:
chore: list accordion design changes

- Update accordion-web CHANGELOG.md.
- Add CHANGELOG.md to atlas-ui with accordion design changes.

diego-antonelli commented on packages/theming/atlas/src/themesource/atlas_core/web/core/helpers/_accordion.scss:
```diff
@@ -0,0 +1,128 @@
+        }
+
+        &.widget-accordion-striped {
+            .widget-accordion-group:not(:is(.widget-accordion-brand-primary, .widget-accordion-brand-secondary, .widget-accordion-brand-success, .widget-accordion-brand-warning, .widget-accordion-brand-danger)):nth-child(2n
```
I think prettier just got confused here

diego-antonelli replied:
Why the stripped is not applied if user defined primary as default color?

IIsaku replied:
According to Chris' requirements, brand styles should override striped styles, since striped styles will apply grid like striped backgrounds to header, which isn't what we want.

diego-antonelli commented on packages/pluggableWidgets/accordion-web/CHANGELOG.md:
```diff
@@ -8,5 +8,6 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 -   Configure groups with a header & content.
```
Isn't changelog here to allows users to see changes? This is a new widget, so I don't think is even necessary a changelog

IIsaku replied:
I think it adds value to add the added features on a high level. The [keep a changelog site](https://keepachangelog.com/en/1.0.0/) also suggests to do this in the example.

diego-antonelli commented on packages/theming/atlas/CHANGELOG.md:
```diff
@@ -0,0 +1,9 @@
+# Changelog
```
This changelog should be for `Atlas Core`

IIsaku replied:
Moved it

IIsaku committed:
chore(atlas-ui): move CHANGELOG.md inside Atlas core
